<a href="https://colab.research.google.com/github/sadiekc/PythonETAVProject/blob/main/PythonETAVProject.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#importing necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
#Loading built-in dataset "tips" from seaborn
tips = sns.load_dataset("tips")

In [ ]:
tips.head(20)

total_bill   tip     sex smoker  day    time  size
0        16.99  1.01  Female     No  Sun  Dinner     2
1        10.34  1.66    Male     No  Sun  Dinner     3
2        21.01  3.50    Male     No  Sun  Dinner     3
3        23.68  3.31    Male     No  Sun  Dinner     2
4        24.59  3.61  Female     No  Sun  Dinner     4
5        25.29  4.71    Male     No  Sun  Dinner     4
6         8.77  2.00    Male     No  Sun  Dinner     2
7        26.88  3.12    Male     No  Sun  Dinner     4
8        15.04  1.96    Male     No  Sun  Dinner     2
9        14.78  3.23    Male     No  Sun  Dinner     2
10       10.27  1.71    Male     No  Sun  Dinner     2
11       35.26  5.00  Female     No  Sun  Dinner     4
12       15.42  1.57    Male     No  Sun  Dinner     2
13       18.43  3.00    Male     No  Sun  Dinner     4
14       14.83  3.02  Female     No  Sun  Dinner     2
15       21.58  3.92    Male     No  Sun  Dinner     2
16       10.33  1.67  Female     No  Sun  Dinner     3
17       16.29  3.71    Male     No  Sun  Dinner     3
18       16.97  3.50  Female     No  Sun  Dinner     3
19       20.65  3.35    Male     No  Sat  Dinner     3

TRANSFORM: Cleanyourdataset.

a. Import your chosen dataset into a JupyterNotebook.

b. Clear out or convert any missing data.

c. Replace or remove any incorrect data.

d. Separate data into any number of appropriate dataframes (groupbys, logical separations, categorical separations,etc.)

In [ ]:
#Checking for missing values that exist in the collection
tips.isnull().sum()

total_bill    0
tip           0
sex           0
smoker        0
day           0
time          0
size          0
dtype: int64

No missing values in this dataset

In [ ]:
#Checking for inconsistent data
tips.describe()

total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000

Minimum value for total bill is 3.07 and for tip column is 1.00 

In [ ]:
#Sorting by the "tip" column in ascending order
tips.sort_values("tip").head()

total_bill   tip     sex smoker  day    time  size
67         3.07  1.00  Female    Yes  Sat  Dinner     1
236       12.60  1.00    Male    Yes  Sat  Dinner     2
92         5.75  1.00  Female    Yes  Fri  Dinner     2
111        7.25  1.00  Female     No  Sat  Dinner     1
0         16.99  1.01  Female     No  Sun  Dinner     2